src/db/oracledb/jobs-dao.js

```diff
@@ -37,23 +120,316 @@ const getJobs = async (internalId, query) => {
   }
 };
 
+/**
+ * Queries for a specific job record belonging to a person
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} internalId Internal ID of a person
+ * @param {string} jobId ID of a job record
+ * @returns {object} Single raw job record
+ */
+const getJobByJobIdWithConnection = async (connection, internalId, jobId) => {
+  const [positionNumber, suffix] = jobId.split('-');
+  const binds = { internalId, positionNumber, suffix };
+  const { rows } = await connection.execute(contrib.getJobs(binds), binds);
+
+  if (rows.length > 1) {
+    throw new Error(`Multiple job records found for job ID ${jobId}`);
+  }
+
+  await getLaborDistributions(connection, internalId, rows);
+
+  return rows[0];
+};
+
+/**
+ * Creates oracledb connection object and calls getJobByJobIdWithConnection
+ *
+ * @param {string} internalId Internal ID of a person
+ * @param {string} jobId ID of a job record
+ * @returns {object} Single raw job record
+ */
 const getJobByJobId = async (internalId, jobId) => {
   const connection = await getConnection('banner');
   try {
-    const [positionNumber, suffix] = jobId.split('-');
-    const binds = { internalId, positionNumber, suffix };
-    const { rows } = await connection.execute(contrib.getJobs(binds), binds);
+    return await getJobByJobIdWithConnection(connection, internalId, jobId);
```

Comment:
Good catch. I did this because I was originally calling it in the post/patch endpoints before I found out all the epafs aren't updated right away. I'll change it back to how it was.